samplesheets/tests/test_permissions_api_taskflow.py

```diff
@@ -74,3 +81,493 @@ def test_get_archive(self):
         ]
         self.assert_response_api(url, good_users, 200, data=self.post_data)
         self.assert_response_api(url, self.anonymous, 401, data=self.post_data)
+
+
+class TestIrodsRequestAPIViewBase(
+    SampleSheetIOMixin, SampleSheetTaskflowMixin, TaskflowAPIPermissionTestBase
+):
+    """Helper mixin for IrodsRequestAPIView tests"""
+
+    def setUp(self):
+        super().setUp()
+        # Import investigation
+        self.investigation = self.import_isa_from_file(SHEET_PATH, self.project)
+        self.study = self.investigation.studies.first()
+        self.assay = self.study.assays.first()
+
+        # Set up iRODS data
+        self.make_irods_colls(self.investigation)
+        self.assay_path = self.irods_backend.get_path(self.assay)
+        self.path = os.path.join(self.assay_path, TEST_FILE_NAME)
+        self.path_md5 = os.path.join(self.assay_path, f'{TEST_FILE_NAME}.md5')
+        # Create objects
+        self.file_obj = self.irods.data_objects.create(self.path)
+        self.md5_obj = self.irods.data_objects.create(self.path_md5)
+
+    def create_request(self):
```

Comment:
Super minor point: I prefer having helper methods before `setUp()`, so that `setUp()` (and possible `tearDown()`) always comes last in the base/helper methods.